src/marks/area.js

```diff
@@ -1,37 +1,22 @@
-import {group} from "d3";
-import {create} from "d3";
-import {area as shapeArea} from "d3";
+import {area as shapeArea, create, group} from "d3";
 import {Curve} from "../curve.js";
 import {defined} from "../defined.js";
-import {Mark, indexOf, maybeColor, titleGroup, maybeNumber} from "../mark.js";
-import {Style, applyDirectStyles, applyIndirectStyles, applyTransform, applyAttr} from "../style.js";
+import {Mark, indexOf} from "../mark.js";
+import {styles, findStyle, applyDirectStyles, applyIndirectStyles, applyTransform, applyGroupedChannelStyles} from "../style.js";
 import {maybeStackX, maybeStackY} from "../transforms/stack.js";
 
 export class Area extends Mark {
-  constructor(
-    data,
-    {
+  constructor(data, options = {}) {
+    const [constants, channels] = styles(options);
+    const {
       x1,
       y1,
       x2,
       y2,
-      z, // optional grouping for multiple series
-      title,
-      fill,
-      fillOpacity,
-      stroke,
-      strokeOpacity,
+      z = findStyle(channels, "fill", "stroke"), // optional grouping for multiple series
```

Comment:
It might be possible to use maybeZ here instead… and then it might be possible to move the call to styles(…) up to the Mark constructor, which would be really nice.

Reply:
Okay, that worked. 😄